packages/react/src/utils/useOnMount.ts

```diff
@@ -0,0 +1,4 @@
+/**
+ * @ignore - internal hook.
+ */
+export { default as useOnMount } from '@mui/utils/useOnMount';
```

Comment:
We're getting rid of @mui/utils dependency (https://github.com/mui/base-ui/pull/827), so let's define our utilities locally.